### PR TITLE
feat(desktop): add slice/task planning detection and slice view

### PR DESCRIPTION
**Pull request by gannonh**
## Summary
- extend planning tool detection to include `kata_create_slice`, `kata_create_task`, and `kata_create_milestone`
- construct slice/task planning artifacts directly from tool args/results and route them through IPC without Linear document fetches
- add `SliceView` and slice planning atom support so slice tabs render description + task checklist with expandable task details
- keep milestone-created ROADMAP refresh on the existing document fetch pipeline
- expand detector and artifact parser test coverage for slice/task/milestone behavior

## Validation
- `cd apps/desktop && bun run typecheck`
- `cd apps/desktop && npx vitest run src/main/__tests__/planning-tool-detector.test.ts src/renderer/lib/__tests__/artifact-parser.test.ts`
- `cd apps/desktop && bun run test`

## Summary by CodeRabbit

* **New Features**
  * Added planning slices, task creation/aggregation, and milestone creation with normalized fields and status mapping.

* **UI**
  * Slice view added to planning pane and artifact tabs; slices ordered and show tasks, statuses, IDs and descriptions.

* **Detection**
  * Artifact detection now recognizes slice title patterns and prefers explicit artifact types.

* **Tests**
  * Expanded tests for document/slice/task/milestone events, payload normalization, edge cases and negative scenarios.

<h3>Greptile Summary</h3>

This PR extends the planning tool detector to handle `kata_create_slice`, `kata_create_task`, and `kata_create_milestone`, routing slice/task artifacts directly through IPC (bypassing Linear document fetches) while keeping milestone creation on the existing document-fetch pipeline. The new `SliceView` component and `slicePlanningAtom` complete the renderer-side rendering of slice tabs with description and expandable task checklist.

<h3>Confidence Score: 5/5</h3>

Safe to merge; all findings are P2 style/robustness suggestions with no current runtime breakage.

The core logic is correct and well-tested. The three findings are all non-blocking: orphaned task handling is a future-proofing concern, the detectArtifactTypeFromTitle divergence has no current impact since slices bypass Linear fetching, and the unanchored regex is a fallback path that is rarely exercised in practice.

ipc.ts (task-dropped edge case and detectArtifactTypeFromTitle divergence); planning-tool-detector.ts (unanchored extractKataId regex)

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/desktop/src/main/planning-tool-detector.ts | Adds detection for kata_create_slice/task/milestone tools with multi-path issue ID extraction and KataId regex fallback parsing; logic is correct and well-covered by tests. |
| apps/desktop/src/main/ipc.ts | Routes slice_created and task_created events directly through IPC without Linear fetches; task_created silently drops if the parent slice artifact is not yet in memory. |
| apps/desktop/src/renderer/atoms/planning.ts | Adds slicePlanningAtom and wires it into applyPlanningArtifactAtom and resetPlanningSessionStateAtom; slice data is kept in sync with planningArtifactsAtom. |
| apps/desktop/src/renderer/components/planning/SliceView.tsx | New read-only slice view with collapsible task checklist; clean implementation with correct readOnly checkbox semantics. |
| apps/desktop/src/renderer/components/planning/ArtifactTabs.tsx | Adds slice to TYPE_SORT_ORDER (position 1) and intentionally omits it from TYPE_LABEL so raw slice titles are displayed; no issues. |
| apps/desktop/src/renderer/lib/artifact-parser.ts | Extends detectArtifactType with three slice-title patterns ([S\d+], S\d+:-, SLICE:) for the renderer; consistent with how the detector emits slice titles. |
| apps/desktop/src/shared/types.ts | Adds PlanningSliceTask, PlanningSliceData, new eventType discriminant, and slice to ArtifactType; types are well-structured and backward-compatible. |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Agent as Kata Agent
    participant PTD as PlanningToolDetector
    participant IPC as ipc.ts (main)
    participant Renderer as Renderer (atoms)

    Note over Agent,Renderer: Slice creation flow
    Agent->>PTD: tool_start (kata_create_slice, args)
    Agent->>PTD: tool_end (kata_create_slice, result { id })
    PTD->>PTD: toPlanningArtifactEvent → slice_created
    PTD->>IPC: emit('artifact', slice_created)
    IPC->>IPC: build PlanningArtifact { artifactType:'slice', sliceData }
    IPC->>Renderer: planningArtifactUpdated (no Linear fetch)

    Note over Agent,Renderer: Task creation flow
    Agent->>PTD: tool_start (kata_create_task, args { sliceIssueId })
    Agent->>PTD: tool_end (kata_create_task)
    PTD->>PTD: toPlanningArtifactEvent → task_created
    PTD->>IPC: emit('artifact', task_created)
    IPC->>IPC: find slice artifact by targetSliceIssueId
    IPC->>IPC: append task to sliceData.tasks
    IPC->>Renderer: planningArtifactUpdated (updated slice)

    Note over Agent,Renderer: Milestone creation flow
    Agent->>PTD: tool_start (kata_create_milestone, args)
    Agent->>PTD: tool_end (kata_create_milestone)
    PTD->>PTD: toPlanningArtifactEvent → milestone_created
    PTD->>IPC: emit('artifact', milestone_created)
    IPC->>IPC: sendPlanningFetchStateToRenderer (start)
    IPC->>IPC: fetchPlanningArtifact (Linear doc fetch for ROADMAP)
    IPC->>Renderer: planningArtifactUpdated (fetched roadmap)
    IPC->>IPC: sendPlanningFetchStateToRenderer (end)
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/desktop/src/main/ipc.ts
Line: 258-265

Comment:
**Task silently dropped if parent slice not yet in memory**

If `task_created` arrives before the corresponding `slice_created` artifact has been stored in `planningArtifactsByKey` (e.g. app reconnect mid-session, or unexpected tool-call ordering), `targetSliceArtifact` will be `undefined` and the task is silently discarded with only a log warning. The task is never retried.

Consider queuing orphaned tasks keyed by `targetSliceIssueId` and flushing them when the matching slice artifact later arrives in `upsertPlanningArtifact`.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/desktop/src/main/ipc.ts
Line: 766-790

Comment:
**`detectArtifactTypeFromTitle` has diverged from `detectArtifactType` in the renderer**

The private helper in `ipc.ts` does not include the `SLICE:` prefix check (`/^SLICE:/`) or the `[S\d+]` / `S\d+:-` patterns that `artifact-parser.ts` now handles. They share the same conceptual purpose but can silently diverge. Since slices bypass Linear fetching this causes no current bug, but a future document fetch that produces a slice-shaped title would be typed as `undefined` instead of `'slice'` in the main process.

Consider re-exporting `detectArtifactType` from `artifact-parser.ts` and using it here, or at minimum adding the slice patterns to the local copy.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/desktop/src/main/planning-tool-detector.ts
Line: 343-356

Comment:
**`extractKataId` regex is unanchored and may match mid-string**

`new RegExp(\`${prefix}\\d+\`, 'i')` matches the *first* occurrence of e.g. `T\d+` anywhere in the title string. A task title like `"Retry T2 implementation"` would extract `T2` rather than failing gracefully. Since `kataId` is passed explicitly by the model in normal usage this is only a fallback concern, but anchoring the search to word boundaries (`\b${prefix}\\d+\b`) would prevent unintended matches.

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["feat(desktop): render slice/task plannin..."](https://github.com/gannonh/kata/commit/950645fc46cfbc19f4b3601166bced39d1f44d8c) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=27232624)</sub>

> Greptile also left **3 inline comments** on this PR.

### CODE REVIEW

File: apps/desktop/src/main/ipc.ts [hunk 1]
```diff
@@ -202,10 +205,92 @@ export function registerSessionIpc({
...
+        )
+      })
+
+      if (!targetSliceArtifact) {
+        log.warn('[desktop-ipc] unable to append task: slice artifact not found', {
+          taskId: planningEvent.task.id,
+          targetSliceIssueId,
+          artifactKey: planningEvent.artifactKey,
+        })
+        return
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/ipc.ts]: Addressed in `68b88953`.

I added orphan-task handling in `apps/desktop/src/main/ipc.ts`:
- queue `task_created` events by `targetSliceIssueId` when no slice artifact is present yet
- flush queued tasks automatically in `upsertPlanningArtifact` when the matching slice artifact appears
- de-duplicate by task id while preserving creation order

This prevents silent task loss during reconnect/reload/order-variance scenarios.

File: apps/desktop/src/main/planning-tool-detector.ts [hunk 2]
```diff
@@ -59,59 +63,187 @@ export class PlanningToolDetector extends EventEmitter {
...
+        action: 'updated',
+        projectId,
+        issueId: sliceIssueId,
+        targetSliceIssueId: sliceIssueId,
+        task: {
+          id: taskId,
+          title: taskTitle,
+          description,
+          status: 'todo',
+        },
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/planning-tool-detector.ts:219]: Fixed in `68b88953`.

`PlanningToolDetector` now derives task status from payloads instead of hardcoding `todo`:
- checks `initialPhase`/`phase`/`status` from tool args and result payloads
- maps to `todo | in_progress | done`

Added coverage in `planning-tool-detector.test.ts` (`maps task status from create_task phase payloads`).

File: apps/desktop/src/main/ipc.ts [hunk 3]
```diff
@@ -202,10 +205,92 @@ export function registerSessionIpc({
...
+      }
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/ipc.ts:314]: Addressed in `68b88953` with the same orphan-task fix:

- unresolved `task_created` events are queued by slice issue id
- queued tasks are merged when the slice artifact is later upserted
- merge is deduped and order-preserving

So tasks are no longer dropped when slice state is missing in-memory.

File: apps/desktop/src/main/ipc.ts [hunk 4]
```diff
@@ -677,3 +762,34 @@ export function registerSessionIpc({
...
+  if (/-CONTEXT(?:\b|$)/.test(normalized) || normalized === 'CONTEXT') {
+    return 'context'
+  }
+
+  if (/^\[S\d+\]\s+/.test(title.trim()) || /^S\d+[:\-\s]/.test(title.trim())) {
+    return 'slice'
+  }
+
+  return undefined
+}
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/ipc.ts:842]: Good catch — fixed in `68b88953`.

I aligned `detectArtifactTypeFromTitle` in `ipc.ts` with slice title handling by adding `SLICE:` detection in the main-process helper, so it no longer drifts from renderer behavior for this pattern.

File: apps/desktop/src/main/planning-tool-detector.ts [hunk 5]
```diff
@@ -175,3 +339,26 @@ function isRecord(value: unknown): value is Record<string, unknown> {
...
+
+  const trimmed = rawValue.trim()
+  const pattern = new RegExp(`${prefix}\\d+`, 'i')
+  const match = trimmed.match(pattern)
+  if (!match) {
+    return undefined
+  }
+
+  return match[0].toUpperCase()
+}
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/planning-tool-detector.ts:357]: Fixed in `68b88953`.

`extractKataId` now only accepts a leading kata token pattern (e.g. `[T01] ...`, `T01: ...`, `T01 - ...`) instead of matching any mid-string `T\d+` fragment.

Also added a regression case in `planning-tool-detector.test.ts` (`Retry T2 implementation details`) to ensure malformed/mid-string fallback IDs are ignored.

File: apps/desktop/src/main/planning-tool-detector.ts [hunk 6]
```diff
@@ -175,3 +340,98 @@ function isRecord(value: unknown): value is Record<string, unknown> {
...
+
+  candidates.push(
+    asString(issuePayload?.status),
+    asString(issuePayload?.phase),
+    asString(issuePayload?.state),
+    asString(statePayload?.name),
+    asString(statePayload?.type),
+    asString(taskPayload?.status),
+    asString(taskPayload?.phase),
+  )
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/planning-tool-detector.ts:392]: [codex] accept — good catch. I will reorder `resolveTaskStatus` precedence so server-returned result fields are evaluated first and args are used only as fallback, then re-run the targeted desktop tests.

[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/planning-tool-detector.ts:392]: [codex] fixed in `be5adb3b`.

- reordered `resolveTaskStatus` candidate precedence so result-derived lifecycle fields are evaluated before request args
- kept `args.initialPhase/phase/status` as fallback only
- added regression coverage (`maps task status from create_task phase payloads`) with a case where args request `executing` but result returns `completed`, now correctly emitted as `done`

Validation:
- `cd apps/desktop && npx vitest run src/main/__tests__/planning-tool-detector.test.ts`
- pre-push hook ran `turbo run lint typecheck test --affected` (green)
